auditing_standards.md

```diff
@@ -237,3 +237,86 @@ Crates with this criteria contain very dangerous unsafe rust code.
     *   Undefined behavior may be possible under common circumstances.
 *   Most crates that try to be sound but don't quite make the cut go in `ub-risk-3`, not here. These crates are wildly unsafe, and the only time we should use them is when they are a necessary evil.
 *   Everything worse than `ub-risk-3` goes in here and we should do our best to avoid using them.
+
+## Criteria-agnostic guidelines
+
+### Delta audits should describe the final version
+
+In general, delta audits should always take the previous, baseline audit into
+account.  In particular, note that
+[Cargo Vet's documentation](https://mozilla.github.io/cargo-vet/audit-entries.html#delta)
+asks the auditor to acquire sufficient context when working on delta
+audits:
+
+> The [auditing] standard here is that the properties are actually
+> preserved, not merely that that the diff doesn't obviously violate
+> them. It is the responsibility of the auditor to acquire sufficient
+> context to certify the former.
+
+Since the delta audits only determine which criteria from the baseline version
+are preserved, you should perform and record a non-delta audit to lower the
+`UB-risk-N` level, or to downgrade from `crypto-safe` to
+`does-not-contain-crypto`. In other words, a delta audit should never result in
+lowering the `UB-risk-N` value from the previous audit, or downgrading
+`crypto-safe` to `does-not-contain-crypto`.
+
+For example, even though
+[the 1.0.79 to 1.0.80 delta of the `proc-macro2` crate](https://chromium-review.googlesource.com/c/chromium/src/+/5453972/2..6)
+doesn't add or change any `unsafe` code, it shouldn't be recorded as
+`ub-risk-0`, because there is still `unsafe` outside of the delta
+(e.g. `unsafe { imp::Literal::from_str_unchecked(repr) }` in
+[`lib.rs` on line 1282](https://chromium-review.googlesource.com/c/chromium/src/+/5453972/2..6/third_party/rust/chromium_crates_io/vendor/proc-macro2-1.0.80/src/lib.rs#1282)).
+
+Another example is the improvements in `flate2` between version 1.0.27
+(`ub-risk-4` because of https://github.com/rust-lang/flate2-rs/issues/220) and
+1.0.28 (`ub-risk-3` because of https://github.com/rust-lang/flate2-rs/pull/373).
+Recording this improvement as a delta audit would be insufficient to pass `cargo
+vet check` presumits (as tested by
+[this ad-hoc Chromium CL](https://crrev.com/c/5532326)).
```

Comment:
@Manishearth, can you PTAL (since this contradicts what you've said earlier)?

In particular, I think the current situation is:

* `vet` sees that `flate2@1.0.27` is `ub-risk-4`
* `vet` sees a delta audit between `flate2@1.0.27` and `flate2@1.0.28` that declares `ub-risk-3` which also implies `ub-risk-4`
* Since `flate2@1.0.27` is *not* `ub-risk-3`, the delta edge doesn't apply / doesn't result in 1.0.28 being treated as `ub-risk-3`

OTOH, I don't have high confidence in the test I've done using the ad-hoc Chromium CL linked from the text above.  Based on my reasoning `ub-risk-4` should be okay, but it still doesn't pass/import... :-/

Reply:
> OTOH, I don't have high confidence in the test I've done using the ad-hoc Chromium CL linked from the text above. Based on my reasoning `ub-risk-4` should be okay, but it still doesn't pass/import... :-/

Oh, this is because when testing I've imported 1.0.30, but the audit only covers 1.0.28.  I've redone my tests (copy&pasting audits from google/rust-crate-audits repo into a local `audits.toml` in Chromium) and I am getting expected results (checks don't pass as `ub-risk-3` before this PR / with delta audit,  and checks pass after this PR / with non-delta audit).

So, this probably deserves a quick review, but I am quite confident that the final/merged state of this PR is correct.